### PR TITLE
Add blog article about JRuby hash implementation

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: _posts/2018-09-04-jruby-hash.markdown [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+So beginning of June I was in contact with [Charles Nutter](https://github.com/headius) and asked for some directions and he suggested to open a pull request. And honestly I didn't like this idea in the beginning as the code was really ugly (I didn't program Java in years) and I also didn't know how much time :clock9: and motivation :sunny: I will have the next weeks to finish the PR. Looking back, I think this was one of the reasons it was eventually a success. Almost immediately after I opened the PR, several JRuby contributors commented on the PR and in IRC to give me feedback and ideas. For instance it was [Marcin Mielżyński's](https://github.com/lopex) idea to get rid of the ``RubyHashEntry`` objects and pack everything into one array. This approach significantly decreased the number of object allocations and improved the performance eventually. Getting as early as possible feedback, even if you only have a prototype, is fundamental!
```
Comment (hennevogel) [inline _posts/2018-09-04-jruby-hash.markdown]: The first time you address somebody by full name it's cool. The second time it feels weird. The third time it's bananas. I would use only the first name after the first full mention...

**Comment by hennevogel** [comment]
Oh an you need review apps!

**Comment by ChrisBr** [comment]
> Oh an you need review apps!

I have review apps, I even sent it to you on IRC :smile: Idk why netlfiy does not comment on the PR but if you click on the checks, you see the netlify URL:

https://deploy-preview-1--sleepy-shannon-3c6917.netlify.com/

**Comment by ChrisBr** [comment]
> Intro
> Hash collisions?
> Chaining and how it looked like
> Open Addressing and how it looks now
> Your experience doing all that shit
> Benchmarks
> ACKs

Thanks for the feedback :heart: I splitted the first paragraph now into two. Can you check if it makes more sense now? Not sure if I understand your structure suggestion because I first show chaining and then open addressing with the problem of chaining in the middle. Not sure if I can move this around because then I would first show the solution before describing the problem, no?

**Comment by ChrisBr** [comment]
> Already way better. Now if you break up the "How hard can it be?" paragraph with a picture/quote in the middle you don't have text desert anymore.

Added some more pictures :+1:

**Pull request merged**